CRM/CiviAwards/Service/AwardProfile.php

```diff
@@ -53,7 +63,25 @@ public function createProfileFields(array $customFields, $profileId) {
     $this->addContactIdField($profileId);
   }
 
-    $this->addContactIdField($profileId);
+  /**
+   * Validates review_field options to ensure legal options are passed.
+   *
+   * @param array
+   *  Custom field
+   *
+   * @throws \CiviCRM_API3_Exception
+   */
+  private function validateReviewFields($customField) {
+    $missingFieldsDiff = array_diff($this->reviewFieldsLegalOptions, array_keys($customField));
+    if (count($missingFieldsDiff) > 0) {
+      $missingFields = implode(', ', $missingFieldsDiff);
+      $exceptionPhrase = (
+        count($missingFieldsDiff) > 1 ? "fields [%1] are" : "field [%1] is"
+      );
+      throw new CiviCRM_API3_Exception(ts("Required $exceptionPhrase missing in review_fields", [
```

Comment:
Can we just throw a normal exception here and not an API exception?
The reason is that the Award profile class may be used internally in code and not just by the API.